### DISCUSSION TITLE
Does Spoke and Hub support any compressed texture format (yet) like KTX2?

### DISCUSSION THREAD

**Discussion by Iaolia456** [mozilla/hubs#4466] [General Questions & Discussion] [answered]
Hi,

I'm starting the optimization phase of my project and I read about compressing texture in gltf file to be in KTX2 container. I found a command line tool calls [gltf-transform](https://github.com/donmccurdy/glTF-Transform) the result file size looks promising. However, I can't import the result model to Spoke, I tried both etc1s and uastc flag but none of them work. It throw error `Cannot read property 'uri' of undefined`, which is pretty not quite helpful.

So, am I missing something? Or Hub just don't support compressed texture format yet?
Sorry, If I sound a bit newbie. I'm not a javascript guy I have limited knowledge about js and anything web-based.

Thanks

**Comment by rawnsley** [comment, marked as answer]
Yes, you can use textures with the [BASISU GLTF extension](https://github.com/KhronosGroup/glTF/blob/master/extensions/2.0/Khronos/KHR_texture_basisu/README.md).

**Reply by Iaolia456** [reply]
I have tried gltf-transform I believe it implement based-on that extension (there is a mention in that readme) but the resulted glb cannot be imported to Spoke. I also try gltfpack but that one throws `unable to encode` warning on all of the texture in the file. I'm finding other glb to try just in case maybe because of my glb file having issue.

**Reply by rawnsley** [reply]
It looks like Spoke isn't supporting BASISU at the moment.  [These two example models](https://github.com/mozilla/hubs/files/6922885/compressed-models.zip) both load in Hubs, but I get errors when I drag them into Spoke.

**Reply by Iaolia456** [reply]
Ah... Ok then I guess no compressed texture format for now then?
